docs/querying/sql-data-types.md

```diff
@@ -40,14 +40,17 @@ milliseconds since 1970-01-01 00:00:00 UTC, not counting leap seconds. Therefore
 timezone information, but only carry information about the exact moment in time they represent. See the
 [Time functions](sql-scalar.md#date-and-time-functions) section for more information about timestamp handling.
 
-The following table describes how Druid maps SQL types onto native types at query runtime. Casts between two SQL types
-that have the same Druid runtime type will have no effect, other than exceptions noted in the table. Casts between two
-SQL types that have different Druid runtime types will generate a runtime cast in Druid. If a value cannot be properly
-cast to another value, as in `CAST('foo' AS BIGINT)`, the runtime will substitute a default value. NULL values cast
-to non-nullable types will also be substituted with a default value (for example, nulls cast to numbers will be
-converted to zeroes).
-
-|SQL type|Druid runtime type|Default value|Notes|
+Casts between two SQL types with the same Druid runtime type (see below table) have no effect, other than exceptions
```

Comment:
There are several important points here. Maybe a list or table would be easier to read?

| case | behaviour |
|------|-----------|
| cast to same Druid runtime type | has no effect (see table below for runtime types and exceptions to this rule) |
| cast to different Druid runtime type | generates a runtime cast |
| value cannot be cast to target type, e.g. `CAST('foo' AS BIGINT)` | use default value of target type or NULL |
| NULL value cast to non-nullable type | use default value of target type |

Reply:
I think we can add this in a followup